engine/src/deploy_macosx.cpp

```diff
@@ -1088,6 +1135,55 @@ static void relocate_function_starts_command(linkedit_data_command *x, int32_t p
 
 ////////////////////////////////////////////////////////////////////////////////
 
+static bool MCDeployToMacOSXFetchMinOSVersion(const MCDeployParameters& p_params, mach_header& p_header, uint32_t& r_version)
+{
+    // First work out what DeployArchitecture to look for.
+    MCDeployArchitecture t_arch;
+    if (p_header . cputype == CPU_TYPE_X86)
+        t_arch = kMCDeployArchitecture_I386;
+    else if (p_header . cputype == CPU_TYPE_X86_64)
+        t_arch = kMCDeployArchitecture_X86_64;
+    else if (p_header . cputype == CPU_TYPE_ARM && p_header . cpusubtype == CPU_SUBTYPE_ARM_V6)
+        t_arch = kMCDeployArchitecture_ARMV6;
+    else if (p_header . cputype == CPU_TYPE_ARM && p_header . cpusubtype == CPU_SUBTYPE_ARM_V7)
+        t_arch = kMCDeployArchitecture_ARMV7;
+    else if (p_header . cputype == CPU_TYPE_ARM && p_header . cpusubtype == CPU_SUBTYPE_ARM_V7S)
+        t_arch = kMCDeployArchitecture_ARMV7S;
+    else if (p_header . cputype == CPU_TYPE_ARM64)
+        t_arch = kMCDeployArchitecture_ARM64;
+    else if (p_header . cputype == CPU_TYPE_POWERPC)
+        t_arch = kMCDeployArchitecture_PPC;
+    else if (p_header . cputype == CPU_TYPE_POWERPC64)
+        t_arch = kMCDeployArchitecture_PPC64;
+    
+    // Search for both the architecture in the mach header and for the 'unknown'
+    // architecture. If the real arch is found, then we use that version; otherwise
+    // if there is an unknown arch then we use that version. If neither are found we
+    // return false which means the caller can do nothing.
+    int t_unknown_index, t_found_index;
+    t_unknown_index = -1;
+    t_found_index = -1;
+    for(uindex_t i = 0; i < p_params . min_os_version_count; i++)
```

Comment:
`min_os_version_count` has type `uint32_t`. But I think it's okay to use `uindex_t i` because we're planning to ensure `sizeof(uindex_t) > sizeof(uint32_t)`.